reader.go

```diff
@@ -0,0 +1,52 @@
+package kinesumer
+
+import (
+	"io"
+
+	k "github.com/remind101/kinesumer/interface"
+)
+
+// Reader provides an io.Reader implementation that can read data from a kinesis
+// stream.
+type Reader struct {
+	// buf is used internally to buffer up received messages.
+	buf []byte
+
+	records <-chan k.Record
+}
+
+// NewReader returns a new Reader instance that reads data from records.
+func NewReader(records <-chan k.Record) *Reader {
+	return &Reader{records: records}
+}
+
+// Read implements io.Reader Read.
+func (r *Reader) Read(p []byte) (n int, err error) {
```

Comment:
I think we should keep track of the last record that we read for each call, and on the next call to read, call `done` on the last one. 

Also, would be nice to have it implement `Closer` and call `done` there as well. 

This will enable this to be used with acknowledgement, which will be pretty cool.